src/screens/Voucher/VoucherDetailScreen.tsx

```diff
@@ -40,10 +46,35 @@ export const VoucherDetailScreen = ({ navigation, route }: StackScreenProps<any>
     fetchPolicy
   });
 
+  const { data: poiVouchersData, refetch: poiVouchersRefetch } = useQuery(
+    getQuery(QUERY_TYPES.POINT_OF_INTEREST),
+    {
+      variables: { id: poiId },
+      skip: !poiId,
+      fetchPolicy
+    }
+  );
+
+  const { vouchers } = poiVouchersData?.[QUERY_TYPES.POINT_OF_INTEREST] ?? {};
+
+  const voucherListItems = useMemo(() => {
+    const parsedListItem = parseListItemsFromQuery(
+      QUERY_TYPES.VOUCHERS,
+      { [QUERY_TYPES.GENERIC_ITEMS]: vouchers },
+      undefined,
+      {
+        withDate: false
+      }
+    );
+
+    return parsedListItem?.filter((item: TVoucherItem) => item?.id !== queryVariables?.id);
+  }, [poiVouchersData]);
```

Comment:
this could be refactored to querying generic items with certain ids as variables, which would return the voucher data directly. additionally the filter method for the current voucher is not necessary, as it will be excluded at querying the server. the voucher generic item ids needs to be passed instead of the `poiId`

Reply:
Can you please check the code again? I tried a different method and I also used `ids` in this method. I think it is easier and more understandable. I'll close this conversation for now.